include/irq.h

```diff
@@ -190,7 +190,12 @@ extern "C" {
  *
  * @return Lock-out key.
  */
+#ifdef CONFIG_SMP
+unsigned int _smp_global_lock(void);
+#define irq_lock() _smp_global_lock()
```

Comment:
wonder if we can use DEPRECATED_MACRO or something to warn people that irq_lock() is no longer the right thing to do.